Linux_Install.sh

```diff
@@ -0,0 +1,227 @@
+
+if [ -z $1 ]
+then
+	echo no option selected - defaulting to single user instalation
+	echo "usage: ./Linux_Install.sh [local] | [ulocal]"
+	echo "        local   - installs IDE for user $USER only ( default if no option given )"
+	echo "        ulocal  - uninstall IDE from user $USER"
+	echo "usage: sudo ./Linux_Install.sh [system] | [usystem]"	
+	echo "        system  - install IDE systemwide for all users"
+	echo "        usystem - uninstall IDE systemwide from all users\n"
+	echo "Both local and system installations may be used on the same computer"
+	echo "for example - a stable release installed systemwide and a nightly for a development user\n"
+
+	MSG="Install IDE 2.0 for user $USER only?"
+
+elif [ $1 = local ]
+then
+		MSG="Install IDE 2.0 for user $USER only?"
+elif [ $1 = ulocal ]
+then
+		MSG="UnInstall IDE 2.0 from user $USER?"
+		MODE=u
+elif [ $1 = usystem ]
+then
+		MSG="UnInstall IDE 2.0 system wide (all users will be affected)?"
+		MODE=s
+elif [ $1 = system ]
+then
+		MSG="Install IDE 2.0 system wide (all users will have access)?"
+		MODE=S
+fi
+	
+# Get absolute path from which this script file was executed
+# (Could be changed to "pwd -P" to resolve symlinks to their target)
+SCRIPT_PATH=$( pwd -P )
+LIB_PATH=$SCRIPT_PATH
+EXE_PATH=$SCRIPT_PATH
+# echo S_PATH = $SCRIPT_PATH
+
+
+read -p "$MSG (Y/N) " YN
+
+if [ -z $YN ]
+then
+	echo OK - exiting
+	exit	
+elif [ $YN = n ]
+then
+	echo OK - No is No... exiting
+	exit
+fi	
+
+echo
+
+###### Remove local user installation
+	
+	if [ $MODE = u ]
+	then
+		echo Uninstalling IDE 2.0 from user $USER
+		echo "Deleting ${HOME}/.local/share/applications/${RESOURCE_NAME}l.desktop"
+		rm   ${HOME}/.local/share/applications/${RESOURCE_NAME}l.desktop
+		
+		echo "Deleting ${HOME}/.local/lib/${RESOURCE_NAME}"
+		rm -rf ${HOME}/.local/lib/${RESOURCE_NAME}
+		
+		echo "Deleting ${HOME}/.local/bin/${RESOURCE_NAME}"
+		rm   ${HOME}/.local/bin/${RESOURCE_NAME}				
+		
+		echo "Installation directory and it's contents including this script can be removed"
+		read -p "Delete directory ${SCRIPT_PATH}? (YES for the affirmative)" YN
+		if [ -z $YN ]
+		then
+			echo exiting; exit
+		elif [ $YN = YES ]
+		then
+			echo "Removing Directory ${SCRIPT_PATH}"
+			cd ..
+			rm -rf ${SCRIPT_PATH}
+		fi
+	fi
+
+###### Perform local user only installation	
+	if [ $MODE = U ]
+	then
+		
+		LIB_PATH=${HOME}/.local/lib/${RESOURCE_NAME}
+		EXE_PATH=${HOME}/.local/bin
+			
+		if [ $YN = Y ] || [ $YN = y ]
+		then
+			mkdir -p $LIB_PATH
+			echo -n "Copying files... "
+			cp -rf * ${HOME}/.local/lib/${RESOURCE_NAME}
+			echo "Files copied...\n"
+		fi
+		
+		mkdir -p "${HOME}/.local/bin"
+		rm -f ${HOME}/.local/bin/${RESOURCE_NAME}
+		ln -s ${HOME}/.local/lib/${RESOURCE_NAME}/arduino-ide ${HOME}/.local/bin/${RESOURCE_NAME}
+	
+		# Create a temp dir accessible by all users
+		TMP_DIR=`mktemp --directory`
+
+		# Create *.desktop file using the existing template file
```

Comment:
the icon and desktop file should follow the proper app-id convention, like the one used for flathub.

See https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s02.html#desktop-file-id

The id should be `cc.arduino.arduinoide2` as suggested for the flathub submission.
Note that if you app register an app-id for DBus use (like Qt or Gtk do) then that's the one that should be used and coordinate with the flathub submission (https://github.com/flathub/flathub/pull/3176 ) to changed it there too.

With that the .desktop should be `cc.arduino.arduinoide2.desktop` and the icon should be `cc.arduino.arduinoide2.png`. Note: if you have a SVG icon it would be so much better.

A 48x48 icon is way too small. 128x128 is the current standard.

Reply:
Just a note the flatpak pr request was updated to change the id to `cc.arduino.IDE2` to match what is used here: https://github.com/arduino/arduino-ide/blob/0a87fd00f38c923a36ce22d9a787fd071a6173e7/electron/packager/config.js#L137

Reply:
then that's what shall be used here too.